client/web/src/site-admin/SiteAdminMigrationsPage.tsx

```diff
@@ -0,0 +1,434 @@
+                titleAtStart={true}
+                defaultExpanded={false}
+            >
+                <div className="pt-2 site-admin-migration-node__errors__grid">
+                    {node.errors.map(error => (
+                        <>
+                            <div className="py-1 pr-2">
+                                <Timestamp date={error.created} now={now} />
+                            </div>
+
+                            <span className="py-1 pl-2 site-admin-migration-node__errors__grid-code">
+                                <code>{error.message}</code>
+                            </span>
+                        </>
+                    ))}
+                </div>
+            </Collapsible>
+        )}
+    </>
+)
+
+/** A major.minor representation of a Sourcegraph version. */
+interface Version {
+    major: number
+    minor: number
+}
+
+/** Extract the major and minor semver numbers from the given version string. */
+const extract = (version: string): Version | undefined => {
+    const semVer = semver.parse(version)
+    if (semVer === null) {
+        return undefined
+    }
+
+    return { major: semVer.major, minor: semVer.minor }
+}
+
+/** Return a copy of the given version with the minor value bumped by delta. */
+const bumpMinor = (version: Version, delta: number): Version => ({ major: version.major, minor: version.minor + delta })
+
+/** Returns true if the given migration state is invalid for the given version. */
+export const isInvalidForVersion = (migration: OutOfBandMigrationFields, version: Version): boolean => {
+    const introduced = extract(migration.introduced)
+    if (introduced && version.major === introduced.major && version.minor < introduced.minor) {
+        return migration.progress !== 0 && !migration.nonDestructive
+    }
+
+    if (migration.deprecated) {
+        const deprecated = extract(migration.deprecated)
+        if (deprecated && version.major === deprecated.major && version.minor >= deprecated.minor) {
+            return migration.progress !== 1
+        }
+    }
+
+    return false
+}
+
+/** Returns the set of migrations that are invalid for the given version. */
+const migrationsInvalidForVersion = (
+    migrations: OutOfBandMigrationFields[],
+    version: Version
+): OutOfBandMigrationFields[] => migrations.filter(migration => isInvalidForVersion(migration, version))
+
+/** Returns true if the given migration is has completed (100% if forward, 0% if reverse). */
+export const isComplete = (migration: OutOfBandMigrationFields): boolean =>
+    (migration.progress === 0 && migration.applyReverse) || (migration.progress === 1 && !migration.applyReverse)
+
+/** Returns the searchable text from a migration. */
+const searchFields = (migration: OutOfBandMigrationFields): string[] => [
+    migration.team,
+    migration.component,
+    migration.description,
+    migration.introduced,
+    migration.deprecated,
+    ...migration.errors.map(error => error.message),
+]
```

Comment:
This fails compilation, maybe this helps?

```suggestion
const searchFields = (migration: OutOfBandMigrationFields): string[] => [
    migration.team,
    migration.component,
    migration.description,
    migration.introduced,
    migration.deprecated,
    ...migration.errors.map(error => error.message),
].filter((value: string | null): value is string => !!value)
```